webbpsf/webbpsf_core.py

```diff
@@ -1323,7 +1323,10 @@ def filter(self, value):
         # long filters have to be used with the CLEARP pupil that contains the
         # PAR reference.
 
-        if self.auto_pupil:
+        if self.auto_pupil and self.filter != 'CLEAR':
+            # note: if the filter is set to 'CLEAR' we have no way of knowing
+            # what the user really wants the pupil mask to be. Could be
+            # NRM, GR700XD, or CLEARP.
```

Comment:
I don't really agree with your comment note. I don't think there is any plan to use NRM+CLEAR  (it would be horribly under sampled at short wavelengths, and probably nearly impossible to reduce the data well.) And likewise CLEAR+CLEARP has no scientific use case (and no usage planned at all so far as I know. Though it shows up in the very-remote-contingency case of turning NIRISS into a replacement third FGS)

You should check with Andre, but I'm not sure there's any case we need to model where CLEAR doesn't go with GR700XD.